### PR TITLE
EDC API Wrapper + EDC API Client Bootstrap

### LINKED ISSUE #187: Bootstrap EDC API Wrapper Extension
# Feature
## Description
Initial required setup so we can start working on implementation in parallel.

## Breakdown
- [x] Java API Client with Dummy API Endpoint from #184 
  - [x] Ensure Dummy API Endpoint is tested
  - [x] Ensure Dummy API Endpoint is documented
  - [x] Ensure public facing Java Client API is well documented
- [x] Add README.md explaining each unit, what it is, and why it exists.
- [ ] Merge into main

### PR DESCRIPTION
**Pull request by richardtreier**
closes #187

### CODE REVIEW

**Review by efiege — COMMENTED** [review]
We should think about separating API's into different extensions. The problem is now, that every EDC, which wants to use the EDC UI API will also expose the other registered API's

**Comment by richardtreier** [comment]
> We should think about separating API's into different extensions. The problem is now, that every EDC, which wants to use the EDC UI API will also expose the other registered API's

We are indeed thinking about ways of doing that.

Our next API Wrapper discussions will probably decide in which way we want to implement code splitting, but for that we need this initial POC Version as subject matter.

File: extensions/wrapper/wrapper-api/build.gradle.kts [hunk 1]
```diff
@@ -47,7 +47,7 @@ tasks.withType<io.swagger.v3.plugins.gradle.tasks.ResolveTask> {
 }
 
 openApiGenerate {
-    generatorName.set("kotlin")
+    generatorName.set("java")
     inputSpec.set("${project.buildDir}/swagger/edcapiwrapper.yaml")
```
Comment (richardtreier) [inline extensions/wrapper/wrapper-api/build.gradle.kts:51]: should be renamed to kebab-case "edc-api-wrapper.yaml"

File: extensions/wrapper/wrapper-api/src/main/java/de/sovity/edc/ext/wrapper/api/ui/kpi/model/KpiResult.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+
+@Getter
+@Setter
+@ToString
+@AllArgsConstructor
+@RequiredArgsConstructor
+@Schema(description = "This is the result of the example endpoint.")
```
Comment (richardtreier) [inline extensions/wrapper/wrapper-api/src/main/java/de/sovity/edc/ext/wrapper/api/ui/kpi/model/KpiResult.java:29]: Wrong docs. E.g. "EDC-status-definining KPIs"

File: extensions/wrapper/client-example/src/main/java/de/sovity/edc/client/examples/GreetingResource.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        query3.setMyNestedList(List.of("a", "b"));
+        query3.addMyNestedListItem("c");
+        return edcClient.exampleApi().exampleEndpoint(query3);
+    }
+
+    @POST
+    @Path("supported-policies")
+    @Produces(MediaType.APPLICATION_JSON)
+    public List<String> getSupportedPolicies() {
+        return edcClient.useCaseApi().getSupportedPolicies();
```
Comment (tmberthold) [inline extensions/wrapper/client-example/src/main/java/de/sovity/edc/client/examples/GreetingResource.java:76]: doesn't compile for me and in the pipeline
image

**Review by tmberthold — DISMISSED** [review]

**Pull request merged**